### PR TITLE
Added hoc import test

### PR DESCRIPTION
**Pull request by 9renpoto**
- Skip hoc props
- Added own prop types

### CODE REVIEW

File: types/react-router-breadcrumbs-hoc/index.d.ts [hunk 1]
```diff
@@ -19,20 +20,14 @@ export interface BreadcrumbsRoute {
...
 
-export interface InjectedProps extends RouteComponentProps<any> {
+export interface InjectedProps extends RouteComponentProps<{}> {
   breadcrumbs: BreadcrumbsProps[];
 }
 
-export type withRouter<P extends RouteComponentProps<any>> = (component: React.ComponentType<P>) => React.ComponentClass<
-    Omit<P, keyof RouteComponentProps<any>>
-  >;
+export type withRouter<P extends RouteComponentProps<any>> = (component: React.ComponentType<P>) => React.ComponentClass<{}>;
```
Comment (9renpoto) [inline types/react-router-breadcrumbs-hoc/index.d.ts]: lost own props 🤔

File: types/react-router-breadcrumbs-hoc/index-tests.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import * as React from 'react';
```
[the remark this answers is not in the document]
Comment (9renpoto) [reply types/react-router-breadcrumbs-hoc/index-tests.tsx]: TypeScript not support `default` imports defaultOptions.
If `allowSyntheticDefaultImports: true`, we use `import React from 'react'`

This PR type definitions with settings recommended by [Microsoft](https://github.com/Microsoft/dts-gen/blob/ee3aa705a65076d394d04745d5fed7d7982a9e19/lib/definitely-typed.ts#L96-L116)

Refs: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/5128#issuecomment-270078424

[the remark this answers is not in the document]
Comment (icd2k3) [reply types/react-router-breadcrumbs-hoc/index-tests.tsx]: Thanks for the explanation!

**Comment by icd2k3** [comment]
@9renpoto this is now released as `2.1.3`. Thanks again!

**Pull request merged**